### PR TITLE
add sdl/2.0.16

### LINKED ISSUE #6804: [request] sdl/2.0.16
### Package Details
  * Package Name/Version: **sdl/2.0.16**
  * Changelog: **https://github.com/libsdl-org/SDL/releases/tag/release-2.0.16**

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.

### PR DESCRIPTION
**Pull request by Croydon**
Specify library name and version:  **sdl/2.0.16**

https://github.com/libsdl-org/SDL/releases/tag/release-2.0.16

Fixes #6804

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jgsogo — APPROVED** [review]

File: recipes/sdl/all/conanfile.py [hunk 1]
```diff
@@ -159,13 +159,6 @@ def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
 
-        # ensure sdl2-config is created for MinGW
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                              "if(NOT WINDOWS OR CYGWIN)",
-                              "if(NOT WINDOWS OR CYGWIN OR MINGW)")
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                              "if(NOT (WINDOWS OR CYGWIN))",
-                              "if(NOT (WINDOWS OR CYGWIN OR MINGW))")
```
Comment (SpaceIm) [inline recipes/sdl/all/conanfile.py:168]: Why dit we ever care to install sd2-config? This file is not relocatable and should be removed from packaging, and we could remove this patch.

File: recipes/sdl/all/conandata.yml [hunk 2]
```diff
@@ -2,7 +2,12 @@ sources:
...
     sha256: d8215b571a581be1332d2106f8036fcb03d12a70bae01e20f424976d275432bc
+  "2.0.16":
+    url: "https://www.libsdl.org/release/SDL2-2.0.16.tar.gz"
+    sha256: 65be9ff6004034b5b2ce9927b5a4db1814930f169c4b2dae0a1e4697075f287b
 patches:
   "2.0.14":
     - patch_file: "patches/0001-fix-cmake-ios-tvos.patch"
       base_path: "source_subfolder"
+    - patch_file: "patches/0002-mingw-improvements.patch"
+      base_path: "source_subfolder"
```
Comment (SpaceIm) [inline recipes/sdl/all/conandata.yml:13]: sdl2-config file is not relocatable and removed from packaging, so I don't think that this patch is required, unless if there was more logic.

Comment (SSE4) [reply recipes/sdl/all/conandata.yml:13]: not sure if it's good idea... I've seen some projects looked for the `sdl2-config` as sdl2 was their dependency (don't remember exact names, but something mpeg related).
maybe we should make `sdl2-config` relocatable instead (or generate one, like we did with freetype)

**Review by SSE4 — APPROVED** [review]

File: recipes/sdl/all/conandata.yml (hunk 2 above)
Comment (SpaceIm) [reply recipes/sdl/all/conandata.yml:13]: It defeats CCI policy

Comment (madebr) [reply recipes/sdl/all/conandata.yml:13]: Perhaps conan needs a generator for creating these config scripts.
These often have a common structure.

File: recipes/sdl/all/conandata.yml [hunk 3]
```diff
@@ -2,7 +2,12 @@ sources:
...
     url: "https://www.libsdl.org/release/SDL2-2.0.14.tar.gz"
     sha256: d8215b571a581be1332d2106f8036fcb03d12a70bae01e20f424976d275432bc
+  "2.0.16":
+    url: "https://www.libsdl.org/release/SDL2-2.0.16.tar.gz"
+    sha256: 65be9ff6004034b5b2ce9927b5a4db1814930f169c4b2dae0a1e4697075f287b
 patches:
   "2.0.14":
     - patch_file: "patches/0001-fix-cmake-ios-tvos.patch"
       base_path: "source_subfolder"
+    - patch_file: "patches/0002-mingw-improvements.patch"
```
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/sdl/all/conandata.yml:12]: the patch for iOS is partially implemented upstream

[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/sdl/all/conandata.yml:12]: No, it's fine for iOS, I don't really understand how CMake goes into the good branch, but it works.

[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/sdl/all/conandata.yml:12]: https://github.com/libsdl-org/SDL/issues/4635

[the remark this answers is not in the document]
Comment (madebr) [reply recipes/sdl/all/conandata.yml:12]: Great detective work :detective:

[the remark this answers is not in the document]
Comment (Croydon) [reply recipes/sdl/all/conandata.yml:12]: Both patch files have a link on top of them to upstream

**Pull request merged**